rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/MtuWatcher.java

```diff
@@ -0,0 +1,45 @@
+package com.polidea.rxandroidble.internal.connection;
+
+
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.inject.Inject;
+import javax.inject.Named;
+import rx.Completable;
+import rx.CompletableSubscriber;
+import rx.Subscription;
+import rx.functions.Action1;
+
+@ConnectionScope
+class MtuWatcher extends Completable implements MtuProvider {
+
+    private final AtomicInteger currentMtuAtomicInteger;
+
+    @Inject
+    MtuWatcher(
+            final RxBleGattCallback rxBleGattCallback,
+            final AtomicInteger atomicInteger,
```

Comment:
What is the reason of injecting an `AtomicInteger` rather than creating it in place? It seems to be only a container for a state. It's more like an implementation detail and should be hidden.

Reply:
It is referenced in the `new OnSubscribe()` and it cannot be referenced there if it is a member of the `MtuWatcher`. The problem is that there is no easy way of extending a `Completable`. Maybe it is even an anti pattern as we can have independent `Completable` subscriptions and shared `AtomicInteger`